app/controllers/reports/regions_controller.rb

```diff
@@ -272,22 +280,72 @@ def monthly_state_data_report
     end
   end
 
-  def monthly_district_report
-    return unless current_admin.feature_enabled?(:monthly_district_report)
+  def hypertension_monthly_state_data
+    @medications_dispensation_enabled = current_admin.feature_enabled?(:medications_dispensation)
+    csv = MonthlyStateData::Exporter.new(
+      exporter: MonthlyStateData::Hypertension.new(
+        region: @region,
+        period: @period,
+        medications_dispensation_enabled: @medications_dispensation_enabled
+      )
+    ).report
+    report_date = @period.to_s.downcase
+    filename = "monthly-district-hypertension-data-#{@region.slug}-#{report_date}.csv"
 
-    @region ||= authorize { current_admin.accessible_district_regions(:view_reports).find_by!(slug: report_params[:id]) }
-    @period = Period.month(params[:period] || Date.current)
-    zip = MonthlyDistrictReport::Exporter.new(@region, @period).export
+    respond_to do |format|
+      format.csv do
+        send_data csv, filename: filename
+      end
+    end
+  end
+
+  def diabetes_monthly_state_data
+    @medications_dispensation_enabled = current_admin.feature_enabled?(:medications_dispensation)
+    csv = MonthlyStateData::Exporter.new(
+      exporter: MonthlyStateData::Diabetes.new(
+        region: @region,
+        period: @period,
+        medications_dispensation_enabled: @medications_dispensation_enabled
+      )
+    ).report
     report_date = @period.to_s.downcase
-    filename = "monthly-district-report-#{@region.slug}-#{report_date}.zip"
+    filename = "monthly-district-diabetes-data-#{@region.slug}-#{report_date}.csv"
 
     respond_to do |format|
-      format.zip do
-        send_data zip, filename: filename
+      format.csv do
+        send_data csv, filename: filename
       end
     end
   end
 
+  def hypertension_monthly_district_report
+    return unless current_admin.feature_enabled?(:monthly_district_report)
+
+    monthly_district_report(MonthlyDistrictReport::Exporter.new(
+      facility_data: MonthlyDistrictReport::Hypertension::FacilityData.new(@region, @period),
+      block_data: MonthlyDistrictReport::Hypertension::BlockData.new(@region, @period),
+      district_data: MonthlyDistrictReport::Hypertension::DistrictData.new(@region, @period)
+    ),
+      @region,
+      @period,
+      diagnosis: :hypertension)
+  end
+
+  def diabetes_monthly_district_report
+    return unless current_admin.feature_enabled?(:monthly_district_report)
+
+    monthly_district_report(
+      MonthlyDistrictReport::Exporter.new(
+        facility_data: MonthlyDistrictReport::Diabetes::FacilityData.new(@region, @period),
+        block_data: MonthlyDistrictReport::Diabetes::BlockData.new(@region, @period),
+        district_data: MonthlyDistrictReport::Diabetes::DistrictData.new(@region, @period)
+      ),
```

Comment:
[suggestion] We spoke about sub-classing these as separate exporter classes from an `Exporter` instead of composition-ing.
**pros**
- the contract between the exporter and the "data" classes is more explicit.
- can share code between `MonthlyDistrictReport::Diabetes` and `MonthlyDistrictReport::Hypertension`

**cons**
- inheriting all this functionality in couples exporting and fetching data from repository